pkgs/by-name/ol/ollama/package.nix

```diff
@@ -184,7 +177,7 @@ goBuild {
   # replace inaccurate version number with actual release version
   postPatch = ''
     substituteInPlace version/version.go \
-      --replace-fail 0.0.0 '${version}'
+      --replace-fail 0.0.0 "$version"
```

Comment:
This is the change that causes rebuild. The rebuild could be prevented with `'${finalAttrs.version}'`, but `$version` is simpler and less hard-coded.

Reply:
This implies, that the top-level attributes are being passed down as environment variables. I think there is an effort to move everything under `env.` to make that explicit, but I do not have a reference right now. I would suggest to move it to `'${finalAttrs.version}'`.

Reply:
> This implies, that the top-level attributes are being passed down as environment variables.

This command is executed in the same Bash shell as other commands in `pkgs/stdenv/generic/setup.sh`. It doesn't have tobe an environment variable to bereferenced here.

Attributes are still passed as shell variables in the `setup.sh`-based build flow when`__structuredAttrs` is true.

Reply:
oh, good to know, thanks. Then it should be good like this.

Reply:
> I think there is an effort to move everything under env.

BTW, I happen to author the PRs that guides all Go environment variables through `env` as the final step to allow `__structuredAttrs = true` for `buildGoModule` (#359641 and #359744).

The progress of #359744 is somewhat stucked, and the main challenge is to finding a way to handle `GOFLAGS` in the new interface. Feedbacks are appreciated!

Reply:
I would use the finalAttrs.version, many packages are already using that. IMO, using the bash variable is not explicit enough to be propagated as a regular practice.

Reply:
That makes sense. I'll change it tonight or tomorrow.

Reply:
Thanks :)